internal/resource/configmap.go

```diff
@@ -63,41 +60,47 @@ func (builder *ServerConfigMapBuilder) Update(object runtime.Object) error {
 	configMap.Labels = metadata.GetLabels(builder.Instance.Name, builder.Instance.Labels)
 	configMap.Annotations = metadata.ReconcileAndFilterAnnotations(configMap.GetAnnotations(), builder.Instance.Annotations)
 
-	if configMap.Data == nil {
-		configMap.Data = make(map[string]string)
+	ini.PrettySection = false // Remove trailing new line because rabbitmq.conf has only a default section.
+	cfg, err := ini.Load([]byte(defaultRabbitmqConf))
+	if err != nil {
+		return err
 	}
+	defaultSection := cfg.Section("")
 
-	var rmqConfBuilder strings.Builder
-
-	_, err := fmt.Fprintf(&rmqConfBuilder, "%scluster_name = %s\n", defaultRabbitmqConf, builder.Instance.Name)
-	if err != nil {
+	if _, err := defaultSection.NewKey("cluster_name", builder.Instance.Name); err != nil {
 		return err
 	}
 
 	if builder.Instance.TLSEnabled() {
-		_, err := rmqConfBuilder.WriteString(defaultTLSConf)
-		if err != nil {
+		if err := cfg.Append([]byte(defaultTLSConf)); err != nil {
 			return err
 		}
 	}
 
 	if builder.Instance.MutualTLSEnabled() {
-		fmt.Fprintf(&rmqConfBuilder, "%s%s\n%s", "ssl_options.cacertfile=/etc/rabbitmq-tls/",
-			builder.Instance.Spec.TLS.CaCertName,
-			defaultMutualTLSConf)
+		if _, err := defaultSection.NewKey("ssl_options.cacertfile", "/etc/rabbitmq-tls/"+builder.Instance.Spec.TLS.CaCertName); err != nil {
+			return err
+		}
+		if _, err := defaultSection.NewKey("ssl_options.verify", "verify_peer"); err != nil {
+			return err
+		}
 	}
 
 	rmqProperties := builder.Instance.Spec.Rabbitmq
-	// rabbitmq.conf takes the last provided value when multiple values of the same key are specified
-	// do not need to deduplicate keys to allow overwrite
-	if rmqProperties.AdditionalConfig != "" {
-		_, err := rmqConfBuilder.WriteString(rmqProperties.AdditionalConfig)
-		if err != nil {
-			return err
-		}
+	if err := cfg.Append([]byte(rmqProperties.AdditionalConfig)); err != nil {
+		return err
+	}
+
+	var buffer bytes.Buffer
+	if _, err := cfg.WriteTo(&buffer); err != nil {
+		return err
+	}
+
+	if configMap.Data == nil {
+		configMap.Data = make(map[string]string)
 	}
 
-	configMap.Data["rabbitmq.conf"] = rmqConfBuilder.String()
+	configMap.Data["rabbitmq.conf"] = buffer.String()
```

Comment:
Small: Rename the `buffer` to maybe `rmqConfBuffer`.

Reply:
Done